src/SlmLocale/Strategy/UriPathStrategy.php

```diff
@@ -193,30 +194,66 @@ public function assemble(LocaleEvent $event)
         $path = $uri->getPath();
 
         // Last part of base is now always locale, remove that
-        $parts = explode('/', trim($base, '/'));
-        array_pop($parts);
+        $parts       = explode('/', trim($base, '/'));
+        $lastElement = count($parts) - 1;
+
+        $removeFirstLocale = true;
+        if (null !== $this->default &&
+            isset($parts[$lastElement]) &&
+            ! in_array($parts[$lastElement], $event->getSupported(), true) &&
+            $parts[$lastElement] !== $this->default
+        ) {
+            $removeFirstLocale = false;
+        }
+
+        if (true === $removeFirstLocale) {
+            // Remove first part
+            array_pop($parts);
+        }
+
         $base  = implode('/', $parts);
 
         if ($base) {
-            $path = substr($path, strlen($base));
+            $path = substr(trim($path, '/'), strlen($base));
         }
-        $parts  = explode('/', trim($path, '/'));
 
-        // Remove first part
-        array_shift($parts);
+        $parts  = explode('/', trim($path, '/'));
+        if (true === $removeFirstLocale) {
+            // Remove first part
+            array_shift($parts);
+        }
 
         if ($locale === $this->default) {
             $locale = '';
         } else {
             $locale .= '/';
         }
 
-        $path = $base . '/' . $locale . implode('/', $parts);
+        $path = ($base ? '/' : '') . trim($base, '/') . '/' . $locale . implode('/', $parts);
         $uri->setPath($path);
 
         return $uri;
     }
 
+    /**
+     * @return SimpleRouteStack
+     */
+    public function getRouter()
```

Comment:
don't add gettters/setters we already have it in the constructor

Reply:
done, i changed it in tests too